File: Sqrt(x)2.cpp
#include<iostream>

using namespace std;

class Solution {
public:
    int mySqrt(int x) {
        if(x==0) return x;
        int left=1, right=x, result;
        while(left<=right){
            int mid=left+(right-left)/2;
            if(mid<=x/mid){
                left=mid+1;
                result=mid;
            }else{
                right=mid-1;
            }
        }
        
        return result;
    }
};

int main(){
    int x;
    cin>>x;
    
    Solution *solution=new Solution();
    cout<<solution->mySqrt(x);
    
    return 0;
}

File: Sqrt(x)2.py
class Solution:
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x==0:
            return x
        left = 1
        right = x
        while(left<=right):
            mid = left+(right-left)//2
            if mid*mid<=x:
                res = mid
                left = mid+1
            else:
                right = mid-1
        return res
        
if __name__=='__main__':
    x = int(input())
    solution = Solution()
    print(solution.mySqrt(x))
